### DISCUSSION TITLE
How to debug genie code?

### DISCUSSION THREAD

**Discussion by monikagujar** [GenieFramework/Genie.jl#513] [Q&A]
I am very new to Julia and, Genie. 

I am using visual studio code, so tried modifying the launch.json file for debugging.

launch.json

`{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "julia",
            "request": "launch",
            "name": "Run Server",
            "program": "${workspaceFolder}/MyApp/bootstrap.jl",
            "args": [
                "s"
            ],
            "stopOnEntry": false,
            "cwd": "${workspaceFolder}/MyApp",
            "juliaEnv": "${command:activeJuliaEnvironment}"
        }
    ]
}`

Genie version = "4.7.1"
julia_version = "1.7.2"

Same issue: https://github.com/GenieFramework/Genie.jl/issues/423

**Comment by AbhimanyuAryan** [comment]
I mostly use Julia's logging: https://docs.julialang.org/en/v1/stdlib/Logging/  instead of Debugger. It's little time consuming then putting breakpoints, watch variables etc but works pretty well

**Comment by essenciary** [comment]
It's definitely worth looking into a step-by-step debugging workflow for Genie now that there are quite a few debugger options available in the Julia ecosystem.

**Reply by essenciary** [reply]
I created an issue for follow up work.
